loop/mi_views.py

```diff
@@ -0,0 +1,189 @@
+
+                mandiobjdetail.distance = row['Mandi Distance'] 
+                mandiobjdetail.category = row['Mandi Category']
+                mandiobjdetail.transport_list.append(transportobj.__dict__)
+
+            mandi_res.append(mandiobjdetail.__dict__)
+
+        # Aggregator Details
+        for aggobj in agg_data_obj:
+            aggobj = AggregatorDetail(aggregator_id=aggobj['user'], aggregator_name=aggobj['name_en'])
+            aggobj.mandi_list = mandi_res
+            agg_res.append(aggobj.__dict__)
+
+        data = json.dumps(agg_res)
+        return HttpResponse(data)
+    else :
+        return HttpResponse(status=401)
+    
+
+def get_crop_prices(request):
+
+    agg_list_requested = is_authenticated(request)
+    
+    if agg_list_requested :
+        # Handling Request Params
+        kwargs = read_params(request)
+
+        cropobj = Crop.objects.values_list('id', flat=True)
+        cropobj = map(int, cropobj)
+        # Prepare data
+        crop_list = tuple(cropobj)
+
+        query = get_query.query_for_rates(crop_list , mandi_list, date_range=kwargs['day_limit'])
+        
+        result = run_query(query)
+        dataframe = remove_crop_outliers(ct_data = result)
+
+        # Grouping Raw Data to Crop & Mandi wise.
+        df = dataframe.groupby(['Crop', 'Market_Real'])
+
+        # Creation of JSON
+        res = []
+        for i, r in df:
+            # Assigning min date
+            latest_price_date = pd.Timestamp('2015-01-01')
+            cropmandidata = CropMandiData(crop_id=i[0], mandi_id= i[1])
+            for index, row in r.iterrows():
+                crop, mandi, date, Av_Rate, STD, PriceMax, PriceMin = row['Crop'], row['Market_Real'], row['Date'], row['Av_Ratemean'], row['STDmean'], row['Pricemax'], row['Pricemin']
+                delta = PriceMax - PriceMin
+                latest_price_date = max(latest_price_date, date)
+                priceobj = PriceDetails(date=str(date), std=round(STD, 2), min_price=round(PriceMin, 2), max_price=round(PriceMax, 2),\
+                                    delta=round(delta, 2), avg_price=round(Av_Rate, 2))
+                cropmandidata.price_details.append(priceobj.__dict__)
+            cropmandidata.latest_price_date = str(latest_price_date)
+            res.append(cropmandidata.__dict__)
+        
+        data = json.dumps(res)
+        return HttpResponse(data)
+    else:
+        return HttpResponse(status=401)
+
+def read_params(request):
+    kwargs = {}
+    
+    kwargs['day_limit'] = int(request.GET.get('day_limit', 3))
+    kwargs['start_date'] = request.GET.get('start_date', None)
+    kwargs['end_date'] = request.GET.getlist('end_date', None)
+    kwargs['crop_id'] = tuple(request.GET.getlist('crop_id', None))
+    kwargs['mandi_id'] = tuple(request.GET.getlist('mandi_id', None))
+    kwargs['range'] = request.GET.get('range', None)
+    kwargs['absolute'] = request.GET.get('absolute', None)
```

Comment:
range & absolute are not self-explanatory. Add comments to explain there meaning